compiler/parser.nim

```diff
@@ -480,6 +483,16 @@ proc dotLikeExpr(p: var Parser, a: PNode): PNode =
   result.add(opNode)
   result.add(a)
   result.add(parseSymbol(p, smAfterDot))
+  if p.tok.tokType == tkParLe and p.tok.strongSpaceA <= 0:
+    var call = newNodeI(nkCall, info)
+    let operator = opNode.ident.s & "()"
```

Comment:
Looks rather hacky. What is it that we're trying to achieve here?

Reply:
It's parsing `a.?b(c)` as `` `.?()`(a, b, c) ``, so that it can be matched by a template or macro.